### ISSUE TITLE
Separate failure cases in aria_hidden_focus_misuse rule for user-enabled and automatic focus

### ISSUE THREAD

**Issue by shunguoy** [IBMa/equal-access#1423]
The current rile only covers one failure case:
   Fail_1: "Element \"{0}\" should not be focusable within the subtree of an element with an 'aria-hidden' attribute with value 'true'"
   This can be confusing when an element (such as area, or summary in a details parent) is automatically focusable by the browser.

  see the test case in test/v2/checker/accessibility/rules/aria_hidden_focus_misuse_ruleunit/aria_hidden_focus_misuse.html
   ![Screenshot 2023-04-27 at 12.47.02 PM.png](https://images.zenhubusercontent.com/61d72f5790898928dfc7024b/da091d4d-cc69-4f4c-ac1c-209af34f6fa6)

**Comment by philljenkins** [comment]
@shunguoy I wanted to confirm that the Help doesn't need to be updated, just the message. 

However, reading the Help I see a note at the bottom of the Help that may be problematic since the browsers **_do_** add `<summary>` (when a child of `<details>`) to the tab order.

> Note: Using aria-hidden="false" on a descendant of an element with `aria-hidden="true"` does not expose that element. Using `aria-hidden="true"` hides itself and all its content from assistive technologies.

screenshot of [aria_hidden_focus_misuse_ Help](https://unpkg.com/accessibility-checker-engine@3.1.47/help/en-US/aria_hidden_focus_misuse.html#%7B%22message%22%3A%22Element%20%5C%22%7B0%7D%5C%22%20should%20not%20be%20focusable%20within%20the%20subtree%20of%20an%20element%20with%20an%20'aria-hidden'%20attribute%20with%20value%20'true'%22%2C%22msgArgs%22%3A%5B%22%7B0%7D%22%2C%22%7B1%7D%22%2C%22%7B3%7D%22%2C%22%7B4%7D%22%5D%2C%22value%22%3A%5B%22VIOLATION%22%2C%22FAIL%22%5D%2C%22reasonId%22%3A%22Fail_1%22%7D):
![image](https://user-images.githubusercontent.com/16193609/234956198-88edb99b-90d8-44cc-959d-f0a96b77a011.png)

**Comment by shunguoy** [comment]
triage: a user can use tabindex='-1' to the element that is automatically focusable (such as summary, area).
